src/utils/useHotKey.tsx

```diff
@@ -4,16 +4,39 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
  */
 
-import { useEffect } from 'react';
+import { DependencyList, useEffect } from 'react';
 import Mousetrap from 'mousetrap';
 
-export default (key: string | string[], onKeypress: () => void) => {
+import { sendUsageData } from './usageData';
+
+export interface Shortcut {
+    hotKey: string;
+    title: string;
+    isGlobal: boolean;
+    action: () => void;
+}
+
+export const shortcuts: Shortcut[] = [];
```

Comment:
Instead of using an array for `shortcuts` it is probably easier to use a `Map`, indexed by the hotkey. With this, removing an entry is just a simple function call instead of having to use `splice`

Reply:
This was one of the first things we discussed when we started implementing shortcuts, but I can't for the life of me remember why we ended up using an array. It may have had something to do with being able to add more shortcuts on the same key, but I agree, this seems like the preferred way to do it.

Reply:
You actually have code, that depends on the assumption, that there are no two shortcuts with the same key:

* This might remove the wrong shortcut if there are two with the same shortcut: https://github.com/NordicSemiconductor/pc-nrfconnect-shared/blob/6c174f450f993f6d984e574f2ae9870fd22cad2e/src/utils/useHotKey.tsx#L31-L36
* Here the same key would be used for React if there are two with the same shortcut:  https://github.com/NordicSemiconductor/pc-nrfconnect-shared/blob/6c174f450f993f6d984e574f2ae9870fd22cad2e/src/Shortcuts/ShortcutModal.tsx#L53-L57